tests/integration/cli/commands/test_issue_2519.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os.path
+import re
+
+import pytest
+
+from pex.compatibility import PY2
+from pex.pip.version import PipVersion, PipVersionValue
+from pex.targets import LocalInterpreter
+from pex.typing import TYPE_CHECKING
+from testing import IntegResults
+from testing.cli import run_pex3
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def assert_lock_sync(
+    lock,  # type: str
+    pip_version,  # type: PipVersionValue
+    *extra_args  # type: str
+):
+    # type: (...) -> IntegResults
+    result = run_pex3(
+        "lock",
+        "sync",
+        "--style",
+        "strict",
+        "--pip-version",
+        str(pip_version),
+        "bloom-filter2>=2.0.0",
+        "--indent",
+        "2",
+        "--lock",
+        lock,
+        *extra_args
+    )
+    result.assert_success()
+    return result
+
+
+TARGET = LocalInterpreter.create()
+PLATFORM_TAG = TARGET.platform.tag
+
+skip_if_py2 = pytest.mark.skipif(
+    PY2, reason="The bloom-filter2 project only claims support for Python 3.x"
+)
+
+
+@skip_if_py2
+@pytest.mark.parametrize(
+    "pip_version",
+    [
+        pytest.param(version, id=str(version))
+        for version in PipVersion.values()
+        if version < PipVersion.v23_2 and version.requires_python_applies(TARGET)
+    ],
+)
+def test_sync_version_mismatch_old_pip(
```

Comment:
This just asserts the OP case explanation offered in https://github.com/pex-tool/pex/issues/2519#issuecomment-2341715425.